db/columns/operations/infer_types.py

```diff
@@ -41,26 +41,43 @@
 }
 
 
-def infer_column_type(schema, table_name, column_name, engine, depth=0, type_inference_dag=None, metadata=None, columns_might_have_defaults=True):
+def infer_column_type(
+        schema,
+        table_name,
+        column_name,
+        engine,
+        depth=0,
+        type_inference_dag=None,
+        metadata=None,
+        columns_might_have_defaults=True,
+):
     """
-    Attempts to cast the column to the best type for it, given the mappings defined in TYPE_INFERENCE_DAG
-    and _get_type_classes_mapped_to_dag_nodes. Returns the resulting column type's class.
+    Attempt to cast the column to the best type for it.
 
     Algorithm:
-        1. reflect the column's type class;
-        2. use _get_type_classes_mapped_to_dag_nodes to map it to a TYPE_INFERENCE_DAG key;
-        3. look up the sequence of types referred to by that key on the TYPE_INFERENCE_DAG;
-            - if there's no such key on the TYPE_INFERENCE_DAG dict, or if its value is an empty
-            list, return the current column type's class;
-        4. iterate through that sequence of types trying to alter the column's type to them;
-            - if the column's type is altered successfully, break iteration and return the output
-            of running infer_column_type again (trigger tail recursion);
-            - if none of the column type alterations succeed, return the current column type's
-            class.
+        0. Check for any data in the column.
+           - If the column is empty, return the column's current type
+             class.
+        0. reflect the column's type class.
+        2. Use _get_type_classes_mapped_to_dag_nodes to map it to a
+           TYPE_INFERENCE_DAG key.
+        3. Look up the sequence of types referred to by that key on the
+           TYPE_INFERENCE_DAG.
+           - If there's no such key on the TYPE_INFERENCE_DAG dict, or if
+             its value is an empty list, return the current column's type
+             class.
+        4. Iterate through that sequence of types trying to alter the
+           column's type to them.
+           - If the column's type is altered successfully, break
+             iteration and return the output of running infer_column_type
+             again (trigger tail recursion).
+           - If none of the column type alterations succeed, return the
+             current column's type class.
```

Comment:
I see you've rewrapped comments from 100 to 80 characters, and changed list's capitalization and punctuation style. I don't have a problem with this style (I don't think 80 chars for docstrings is enoug though), but I think we should postpone any such formatting until we have a style/code standard. A notable exception is if a given style is significantly inconsistent, but in this case this was consistent with all of my other comments. Do you agree about postponing unurgent comment style changes?

A basic problem here is that it's bloated the diff and I'm not sure if I'm seeing all the meaningful changes I'm meant to see.

Reply:
Not requesting changes regarding style, btw.

Reply:
To be honest, I did some punctuation changes for my own readability, and my editor did the rewrapping as per PEP8 on lines I'd changed.

Reply:
> A basic problem here is that it's bloated the diff and I'm not sure if I'm seeing all the meaningful changes I'm meant to see.

This is a valid complaint.

Reply:
If your editor didn't automatically reformat each comment or docstring you edit, that would improve the diffs.